sacred/utils.py

```diff
@@ -315,6 +315,57 @@ def __init__(
         super().__init__(message, print_traceback, filter_traceback, print_usage)
 
 
+class Traceback:
+    def __init__(self, tb_frame, tb_lasti, tb_lineno, tb_next):
+        self.tb_frame = tb_frame
+        self.tb_lasti = tb_lasti
+        self.tb_lineno = tb_lineno
+        self.tb_next = tb_next
+
+
+class FilteredTracebackException(tb.TracebackException):
+    def __init__(
+        self,
+        exc_type,
+        exc_value,
+        exc_traceback,
+        *,
+        limit=None,
+        lookup_lines=True,
+        capture_locals=False,
+        _seen=None,
+    ):
+
+        filtered_exc_traceback = []
+        tb = exc_traceback
+        while tb is not None:
+            if not _is_sacred_frame(tb.tb_frame):
+                filtered_exc_traceback.append(
+                    Traceback(tb.tb_frame, tb.tb_lasti, tb.tb_lineno, tb.tb_next)
+                )
+                if len(filtered_exc_traceback) >= 2:
+                    filtered_exc_traceback[-2].tb_next = filtered_exc_traceback[-1]
+            tb = tb.tb_next
```

Comment:
This currently has the same problem as before: It ignores the traceback of chained exceptions in `exc_value.__cause__.__traceback__`. It should be possible to simply overwrite `exc_value.__cause__.__traceback__`, so this could be a solution.

Reply:
It also ignores `exc_value.__context__.__traceback__`

Reply:
So, I see two solutions here:
1. Overwrite `exc_traceback`, `exc_value.__context__.__traceback__` and `exc_value.__cause__.__traceback__` with a filtered version, or
2. Set the `stack` attribute of `self`, `self.__cause__` and `self.__context__` with a filtered `StackSummary`.

Not sure yet which one I like better, but the first one doesn't involve computing multiple `StackSummary` instances, so I guess it's the better solution.

Reply:
The same filtering should be possible, I'll have look.

Reply:
Added a test for a chained error and added the filtering.